akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -227,4 +228,10 @@ object EventSourcedBehavior {
    * By default, snapshots and events are recovered.
    */
   def withRecovery(recovery: Recovery): EventSourcedBehavior[Command, Event, State]
+
+  /**
+   * Publish events to a topic after they have been persisted
+   */
+  @ApiMayChange
+  def withEvenPublishing(topicName: String): EventSourcedBehavior[Command, Event, State]
```

Comment:
The idea is that the topic name would be node-unique, with a local replicator forwarding across sharding. But it also opens up for one replicator in the entire "DC" using the topic in a distributed way, and potentially using speculative replication even when not using sharding (one AA entity per node in a cluster for example).

Tricky part will be how to allow passing the PublishedEvent to the entity `ActorRef[Command]` in those cases.

Reply:
After some discussion we decided that it is better (lower overhead) to publish this to the system local event bus, if we want to cluster-pub-sub it we could then do that from a subscriber later.